projects/pgai/pgai/vectorizer/vectorizer.py

```diff
@@ -717,6 +742,8 @@ async def _copy_embeddings(
             conn (AsyncConnection): The database connection.
             records (list[EmbeddingRecord]): The embedding records to be copied.
         """
+        if self.copy_types is None:
+            await self._load_copy_types(conn)
```

Comment:
This duplicates logic in the block below. Either move this there, or remove the whole `if self.copy_types is None` below.

Reply:
whoops. doh.